File: ruby/0058-length-of-last-word.rb
# @param {String} s
# @return {Integer}

#simple one liner
def length_of_last_word(s)
  s.split.last.length
end

#double pointer
def length_of_last_word(s)
    left = -1
    while s[left] == ' ' do
        left -= 1
    end

    right = left
    while s[left] && s[left] != ' ' do
        left -=1
    end

    right - left
end

File: python/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        """
	one shortcut
	"""
	#	return len(s.split()[-1])
        count = 0
	for i in range(len(s) - 1, -1, -1):
		char = s[i]
		if char == " ":
			if count >= 1:
				return count
		else:
			count += 1
	return count
